scripts/corelli/calibration/database.py

```diff
@@ -211,29 +309,129 @@ def combine_spatial_banks(tables:WorkspaceGroup, name: Optional[str] = None)->Ta
         assert isinstance(table, TableWorkspace), message
 
         # check column names
-        if not has_valid_columns(table):
-            raise RuntimeError('Table index ' + str(i) + ' is not a valid Corelli TableWorkspace ["Detector ID", "Detector Position"]')
+        if not has_valid_columns(table, table_type=table_type):
+            message = f'Table index {i} of type {table_type }is not a valid Corelli TableWorkspace'
+            raise RuntimeError(message)
 
         table_dict = table.toDict()
 
         # loop through rows
-        for r in range(0,table.rowCount()):
-            combined_table.addRow( [table_dict['Detector ID'][r], table_dict['Detector Position'][r] ])
+        column_names = table.getColumnNames()  # 'Detector ID' and 'Detector Y Coordinate' for 'calibration' table_type
+        for r in range(0, table.rowCount()):
+            combined_table.addRow([table_dict[name][r] for name in column_names])
 
     return combined_table
 
 
-def verify_date_YYYYMMDD(function_name:str, date:str)->None:
+def day_stamp(input_workspace: InputWorkspaceTypes) -> int:
+    r"""
+    Find the day YYYYMMDD stamp (e.g 20200311 for March 11, 2020) using metadata from the
+    Nexus events file as input.
+
+    :param input_workspace: input workspace from which the day stamp is to be retrieved.
+    """
+    run = mtd[str(input_workspace)].getRun()
+    for log_key in ('run_start', 'start_time'):
+        if log_key in run.keys():
+            return int(run[log_key].value[0:10].replace('-', ''))
+    raise KeyError(f'Day stamp not found in the metadata of {str(input_workspace)}')
+
+
+def verify_date_format(function_name: str, date: Union[int, str]) -> None:
     """
     Function that verifies if date input string has a correct date format: YYYYMMDD.
     Throws an exception otherwise.
-    :param function_name input that calls this funciton for better exception handling
+    :param function_name input that calls this function for better exception handling
     :param date input date string to be verified
     :raises ValueError if date is not in YYYYMMDD format
     """
     try:
-        if len(date) != 8:
+        if len(str(date)) != 8:
             raise ValueError('')
-        datetime.strptime(date, '%Y%m%d')
+        datetime.strptime(str(date), '%Y%m%d')
     except ValueError:
-        raise ValueError('date in function ' + function_name + ' ' + date +  ' is not YYYYMMDD')
+        raise ValueError('date in function ' + function_name + ' ' + str(date) + ' is not YYYYMMDD')
+
+
+def new_corelli_calibration(database_path: str,
+                            date: Optional[str] = None) -> Tuple[str, str, str]:
+    r"""
+    Generate a Corelli calibration set of files for a given day stamp, or for today if no day stamp is given.
+
+    For each bank, this function will retrieve the calibration with an anterior date as close as possible
```

Comment:
Similarly here, add an example (with path from the database base directory) of the the expected output in the database when running this new function.

Reply:
Done